### PR TITLE
Spelling and grammatical errors

### PR DESCRIPTION
**Pull request by madhavshah49**

### CODE REVIEW

**Comment by birm** [comment]
Is this a duplicate of #256 ?

**Comment by madhavshah49** [comment]
> Is this a duplicate of #256 ?

There was an mistake in that PR , the licence.txt remains the same while I have made some additional changes in the Readme file

File: LICENSE.txt [hunk 1]
```diff
@@ -26,7 +26,7 @@ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
 ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER, CAUSED AND ON
```
Comment (birm) [inline LICENSE.txt:29]: I would suggest against editing licenses in this way, at least here. Licenses are specifically written to convey certain legal rights and protections. If you feel that a license doesn't read well, I'd contribute upstream to the license template itself. You can find a human-readable summary of bsd-3 [here](https://tldrlegal.com/license/bsd-3-clause-license-(revised)), and the source of the boost license seems to be [here](https://www.boost.org/LICENSE_1_0.txt).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -98,9 +98,9 @@ vcpkg install ensmallen:x64-windows
 
 ### Build and Test
 
-This section describes how to build the **ensmallen** tests from source. **ensmallen** uses CMake as its build system and [Catch2](https://github.com/catchorg/Catch2) as the unit test framework.
+This section describes how to build the **ensmallen** tests from the source. **ensmallen** uses CMake as its build system and [Catch2](https://github.com/catchorg/Catch2) as the unit test framework.
 
-First, clone the source code from Github and change into the cloned directory. Or alternatively, you can download the latest relese from the [website](http://ensmallen.org) and extract it.
+First, clone the source code from Github and change it into the cloned directory. Alternatively, you can download the latest release from the [website](http://ensmallen.org) and extract it.
```
Comment (birm) [inline CONTRIBUTING.md:103]: I think that "change into so-and-so directory" is reasonably understood, and I'd strongly recommend against adding "it" to this sentence.
```suggestion
First, clone the source code from Github and change into the cloned directory. Alternatively, you can download the latest release from the [website](http://ensmallen.org) and extract it.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,20 +15,20 @@ IRC channel on irc.freenode.net.
...
 functionality, that you've added some tests in the `tests/` directory.  And if
-you are fixing a bug, it's always nice to include a test case if possible to so
+you are fixing a bug, it's always nice to include a test case if possible so
 that the bug won't happen again.
 
 ## Build/test process
 
 All of the code for ensmallen is located in `include/ensmallen_bits/` and all of
-the tests are in `tests/`.  ensmallen is header-only, so anything in
+The tests are in `tests/`.  ensmallen is header-only, so anything in
```
Comment (birm) [inline CONTRIBUTING.md:31]: This is a continuation of the previous sentence, don't capitalize here.
```suggestion
the tests are in `tests/`.  ensmallen is header-only, so anything in
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -113,7 +113,7 @@ $ tar -xvzpf ensmallen-2.16.0.tar.gz
 $ cd ensmallen-latest
 ```
 
-Next, make a build directory and change into that directory.
+Next, make a build directory and change it into that directory.
```
Comment (birm) [inline CONTRIBUTING.md:116]: As above, I don't think "it" adds clarity here, and possibly makes it less clear.
```suggestion
Next, make a build directory and change into that directory.
```

**Review by birm — CHANGES_REQUESTED** [review]
This is an interesting PR. I appreciate efforts to increase readability, and I'm sure the rest of the community does too. However, this is a bit of a mixed bag. I've tried to give some comments, and I hope they're helpful.

File: LICENSE.txt (hunk 1 above)
Comment (madhavshah49) [reply LICENSE.txt:29]: thank you for that information , I shall not make changes I shall not make any changes in the license agreement

**Comment by madhavshah49** [comment]
@birm Thank you for pointing out those changes , I've made the changes

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -15,14 +15,14 @@ IRC channel on irc.freenode.net.
 Once a pull request is submitted, it must be reviewed and approved before a
 merge, to ensure that:
 
- * the design matches the rest of the ensmallen design
+ * The design matches the rest of the ensmallen design
```
Comment (birm) [inline CONTRIBUTING.md:18]: The other bullet points aren't capitalized, for consistency:
```suggestion
 * the design matches the rest of the ensmallen design
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -98,9 +98,9 @@ vcpkg install ensmallen:x64-windows
 
 ### Build and Test
 
-This section describes how to build the **ensmallen** tests from source. **ensmallen** uses CMake as its build system and [Catch2](https://github.com/catchorg/Catch2) as the unit test framework.
+This section describes how to build the **ensmallen** tests from the source. **ensmallen** uses CMake as its build system and [Catch2](https://github.com/catchorg/Catch2) as the unit test framework.
```
Comment (zoq) [inline CONTRIBUTING.md:101]:
```suggestion
This section describes how to build the **ensmallen** tests from source. **ensmallen** uses CMake as its build system and [Catch2](https://github.com/catchorg/Catch2) as the unit test framework.
```

I would prefer to go with "from source", unless you like to write "from the source code", but "from source" is more common.

**Review by rcurtin — APPROVED** [review]
Sure, personally I think these are reasonable changes.  Thanks @madhavshah49. :+1:

**Review by birm — APPROVED** [review]
Looks good to me, thanks for accepting my suggestions.

**Comment by zoq** [comment]
> @rcurtin @zoq @madhavshah49 @birm It's a **really bad idea** to modify the license file. This is no longer the the 3-clause BSD license.
> Little changes may seem benign, but they may have massive implications from a legal point of view.
> 
> This change in particular raises red flags:
> (original) `LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON`
> (changed) `LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER, CAUSED AND ON`
> 
> The extra comma after "HOWEVER" changes meaning of the sentence.
> 
> I strongly suggest to revert to the original license, and never touch this file again.

Totally agreed, but looks like the merged changes do not contain anything related to the license file, unless you mean the commit history?

**Comment by birm** [comment]
My bad, I absolutely should have squashed the commits on merge. I can rebase them now if we're worried, just let me know.

**Comment by madhavshah49** [comment]
My fault, if anything I can make the changes

**Pull request merged**